src/mrmailbox.h

```diff
@@ -346,7 +355,11 @@ void            mrmailbox_log_error         (mrmailbox_t*, int code, const char*
 void            mrmailbox_log_error_if      (int* condition, mrmailbox_t*, int code, const char* msg, ...);
 void            mrmailbox_log_warning       (mrmailbox_t*, int code, const char* msg, ...);
 void            mrmailbox_log_info          (mrmailbox_t*, int code, const char* msg, ...);
+#ifndef PY_CFFI
 void            mrmailbox_log_vprintf       (mrmailbox_t*, int event, int code, const char* msg, va_list);
+#else
+void            mrmailbox_log_vprintf       (mrmailbox_t*, int event, int code, const char* msg, ...);
+#endif
```

Comment:
Figure out what's up with va_list.  AFAIK the `...` notation is also valid C but it's what CFFI can parse.  This results in a compilation warning though:
```
warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
   *(void(* *)(mrmailbox_t *, int, int, char const *, ...))o = mrmailbox_log_vprintf;
```

Reply:
maybe the non-`...` notation also works for the other systems; we can then switch completely.

Reply:
I'm not sure I understand your response here.  I think the standard C-notation for a function with variadic arguments is `f(int named, ...)`.  If I understand correctly you pass in va_list as a lone type declaration in the signature which afaik is not standard C.  So is there a reason to do this instead of the standard `...` notationn?  It would be nice if all code could use the standard notation for variadic function arguments instead of having to ifdef for cffi like this.  What would you lose if you moved all varargs to this?

Reply:
> So is there a reason to do this instead of the standard ... notationn? 

Yes, the `...`-notation does not allow to pass a variable argument list to another function. The solution for this is to convert the list into an object using va_start() and pass this to other functions. 

However, there seems to be an easy solution for us: I think I can make mrmailbox_log_vprintf() private so that it is not seen by CFFI.

NB: I am currently on the way moving all structures to internal files as `struct _mrchat { ... }` the public headers use a `typedef struct _mrchat mrchat_t;` then.